mnet/mnet_test.go

```diff
@@ -52,6 +54,42 @@ func TestNetwork(t *testing.T) {
 		}
 	})
 
+	t.Run("ListenRandom", func(t *testing.T) {
+		synctest.Test(t, func(t *testing.T) {
+			n := mnet.New(t.Name())
+			lst, err := n.Listen("tcp", "foo:0")
```

Comment:
Maybe do this a few times so we have some assurance we're getting distinct addresses. I suspect this test would have passed previously too.